app/assets/javascripts/index.js.erb

```diff
@@ -0,0 +1,99 @@
+$( document ).ready(function() {  
+  // Clear the facilities list
+    $("#facilitiesList").html("");
+  $('#services_selection').hide();
+  
+  L.mapbox.accessToken = 'pk.eyJ1IjoiYXlhbGVsb2VociIsImEiOiJjaWtmdnA1MHAwMDN4dHdtMnBqbGR3djJxIn0.fuqVOKCu8mE-9IdxTa4R8g';
+  var map = L.mapbox.map('map', 'mapbox.streets')
+        .setView([40, -74.50], 5);
+  map.scrollWheelZoom.disable();
+  
+  // Get user location from HTML5 location API if available
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(recenter, function() {});
+  }
+  
+  $('#search').on("click", function(){   
+    var classification = $('select[id=facilities_selection]').val();
+    
+    $.ajax({
+        url: '/search/show',
+        type: 'GET',
+        data: {'Classification': classification}, 
+        success: function(data){
+            plotData(data, map);
+        },
+        failure: function(data){
+            alert(data);
+        }
+    });
+  });
+  
+  function recenter(position) {
+    map.setView({
+      lat: position.coords.latitude,
+      lng: position.coords.longitude
+    });
+    map.setZoom(12);
+  }
+  
+  $('#facilities_search').on("click", function(){    
+        $('#services_selection').hide();
+        $('#facilities_selection').show();
+    });
+    
+  $('#services_search').on("click", function(){
+        $('#facilities_selection').hide();
+        $('#services_selection').show();
+    });
+    
+    //plot data on the map
+    function plotData(data, map){
+      //remove points from map
+      
+      //remove facilities from list
+      $("#facilitiesList").html("");
+      
+     for (var i = 0; i < data.length; i++) {
+       if (isNaN(parseFloat(data[i]['Longitude'])) || isNaN(parseFloat(data[i]['Latitude']))) {
+         continue;
+       }
+       
+        L.mapbox.featureLayer({
+          // this feature is in the GeoJSON format: see geojson.org
+          // for the full specification
+          type: 'Feature',
+          geometry: {
+              type: 'Point',
+              // coordinates here are in longitude, latitude order because
+              // x, y is the standard for GeoJSON and many formats
+              coordinates: [ data[i]['Longitude'], data[i]['Latitude']]
+          },
+          properties: {
+            title: data[i]['Location_Descriptive_Name_Common_Name'],
+            radius: 6,
+            fillColor: 'white',
+            color: '#1B384E',
+            weight: 2,
+            opacity: 1,
+            fillOpacity: 1
+          }
+        }).addTo(map);
```

Comment:
You'll probably need to keep a list/array of markers/pins added so that you can clear the map. The `//remove points from map` area can then iterate over that list and clear them using `map.removeLayer(marker)`. (Disclaimer: My Mapbox/Leaflet knowledge is circa 2012, so there might be a fancier way of doing this nowadays. :smile: )